docs/conf.py

```diff
@@ -236,7 +235,7 @@ def generate_r_docs(app):
     /home/docs/.conda/bin/conda install -q -y -n r_env -c conda-forge \
         r-data.table=1.12.8=r35hcdcec82_0 \
         r-pkgdown=1.3.0=r35h6115d3f_1001 \
-        r-roxygen2=6.1.1=r35h0357c0b_1001
+        r-roxygen2=7.1.0=r36h0357c0b_1
```

Comment:
@jameslamb 

Build hangs due to `conda` cannot resolve this issue: `r-base=3.5` and `r-roxygen2=r36...`

> I'm also proposing bumping up our version of `roxygen2` to 7.1.0, the most recent version used in `DESCRIPTION` 

I remember it took me several days to find out the concrete versions of packages that do not conflict with each other and do not lead to the error about that DLL could not be loaded (one famous upstream `pkgdown` issue unfortuanately I cannot find references to). I believe that bumping roxygen2 version can be done in a separate PR or together with #3036 (most likely that we will have to bump all versions due to 1 change).

Reply:
hmmmm actually I think there is so much work involved in doing this, it might be worth it for me to take on #3036 within this PR.

Reply:
> hmmmm actually I think there is so much work involved in doing this, ...

Yeah, indeed. So maybe delay this for later time and concentrate on CRAN release? I mean, docs are run OK right now and this is not something user-related.

Reply:
I don't want to pick back up the next big CRAN step (https://github.com/jameslamb/LightGBM/pull/15 ) while #2965 is awaiting review from R reviewers, since I really need the CI coverage and speedups there to move forward with confidence.

So until we get that merged, I'm spending some time on these other side things.

Reply:
Ah, I see, you are full of power! Great! 🎉